packages/block-library/src/query/edit.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Internal dependencies
+ */
+import QueryPanel from './query-panel';
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { debounce, isUndefined, pickBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockList,
+	BlockEditorProvider,
+	InspectorControls,
+	WritingFlow,
+} from '@wordpress/block-editor';
+import { cloneBlock } from '@wordpress/blocks';
+import { PanelBody, Placeholder, Spinner } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { EntityProvider } from '@wordpress/core-data';
+import { withSelect } from '@wordpress/data';
+
+class Edit extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			editingPost: null,
+			blocksTree: {},
+		};
+
+		this.debouncedCreateBlockTree = debounce(
+			this.createBlockTree.bind( this ),
+			1000
```

Comment:
While testing I noticed that my changes take a lot of time to propagate to the other blocks, Is this one-second delay necessary?

Reply:
It seems to be, because changes to block content / attrs will propagate and interfere with the usability. 

The easiest way to see the problem is to remove the debounce, add a paragraph block inside the Query, and start typing. When you are typing, the block tree will keep getting updated and the updates will cause your typing to lag, and in some cases not register at all.